### ISSUE TITLE
howto use setcap to a snap installed dosbox-x?

### ISSUE THREAD

**Issue by Bedenk** [joncampbell123/dosbox-x#3432]
### Code of Conduct & Contributing Guidelines

- [X] I agree to follow the code of conduct and the contributing guidelines.

### Have you checked that no other similar question(s) already exists?

- [X] I have searched and didn't find any similar issues.

### Question

As the title say, how do I get setcap set on a dosbox-x executable if it is installed by snap on Ubuntu 22.04?
I use "sudo setcap cap_net_raw+ep /snap/dosbox-x/840/dosbox-x" and I get "Failed to set capabilities on file `/snap/dosbox-x/840/dosbox-x' (Read-only file system)".

How to get a network going in dosbox-x without the setcap methode, or how to pass the setcap methode to a file hostage by snap?

Someone here to help out? Thx in advance, Pit

**Comment by Bedenk** [comment]
Oke, thx. Yes you are right @rderooy. I have now build a version (38.24) from scratch and it works. But still no network.
I think something went wrong with setcap. In my log I have "SDLNet_TCP_Open: Couldn't connect to remote host".

Pit

**Comment by rderooy** [comment]
Try to run `getcap` against the binary to validate.

```
getcap /usr/bin/dosbox-x
/usr/bin/dosbox-x cap_net_raw=ep
```

If that returns the same thing, it must be something else. What software are you using inside dosbox-x?

**Comment by Bedenk** [comment]
Hit thx for the help.
I have windows 98 running. Besides the networking and the use of the dos prompt it is running fine.
ne2000  = true
nicbase = 300
nicirq  = 10
backend = pcap
realnic = 1 (This number came from list)

Thats it.

**Comment by Bedenk** [comment]
My log file after i started my session with "/usr/bin/dosbox-x cap_net_raw=ep".
[dosbox-x.log](https://github.com/joncampbell123/dosbox-x/files/8589035/dosbox-x.log)

**Comment by rderooy** [comment]
Just to confirm, when you run `getcap`, you get the same output as I posted above? cap_net_raw=ep is not a parameter to the dosbox-x binary.

From your log, it ends up using `enp4s0` as the network device. That is a wired ethernet connection? (it will not work with Wifi).

FYI: For the `realnic=` option, you should really use the device name (e.g. "enp4s0") and not the index from the list. If you plug/unplug any network devices the index may change.

**Comment by rderooy** [comment]
Also I think the "SDLNet_TCP_Open" error is a [red herring](https://en.wikipedia.org/wiki/Red_herring), I think it has to do with TiMidity right after it, and has nothing t do with your actual problem.

**Comment by Bedenk** [comment]
Yes, with getcap i get exactly the same output as you wrote. And indeed the enp4s0 is a wired connection.

**Comment by Bedenk** [comment]
I found one problem, the nic got irq 3 from windows but the config say irq 10. So I changed that in the windows configuration and now I see in my dhcp server that it gets a address.
Within windows i still can't do a thing with it.

**Comment by rderooy** [comment]
If you get an address from the DHCP server, you should be fine. You should be able to reach other devices on the network. But note that old web browsers are useless at opening modern websites.

**Comment by Bedenk** [comment]
Thx @rderooy. I got things a bit going.... I can ping! Yes I can :-)
But The pings are very slow and IE can't reach a site, noting. And I checked, it has nothing to do with dns. Same reaction if I try to reach a site using its IP.
I can't explore the inside network. So I have to do something with Windows.
I did a fresh install of Windows and used a 2 gig drive now, the dos prompt is working normal. So I can use ipconfig.
I still have 2 NIC's. So I think I start removing one of them.

**Comment by Bedenk** [comment]
I have installed a webserver on my host pc to serve a simpel htm file. It can't reach it. I can't ping windows 98. My host pc has 192.168.0.37 and the win98 pc has 192.168.0.59. So it has to be simple to ping the son of a b......

**Comment by rderooy** [comment]
Your network switch may also be causing issues. Some switches only allow a single MAC address per port (typically these are the more advanced switches where this is configurable).

But I just tried it here with PCAP and am also having problems.
- ping outside my network fails
- ping between host and dosbox-x fails
- ping to other devices on my home network "works", but I loose 50-75% of packets

**Comment by rderooy** [comment]
Using the slirp backend seems to work better with regards to pings.

**Comment by Jookia** [comment]
Do we need an issue about the pcap backend being unstable? It might be worth looking closer at that.

**Comment by Bedenk** [comment]
When it is the normal way to use, in my case it didn't function. So yes maybe it is a good idea to op a ticket?
But how and where?

**Comment by Bedenk** [comment]
I wanted to use win98 to view Gopherspace. Netscape is indeed the tool to do so. But in the meantime I installed a version of Heiku on VirtualBox (bridge adapter) and guess what? WebPositive (default webbrowser of Heiku) can read gopher pages and make use of https at the same time. (https is someting Netscape can't use because Netscape is older then https). And with Heiku I could make a NFS mount and edit al my Gopherspace material in Heiku. So, for me, I don't need to dive deeper in dosbox-x for now. The only thing is, I love operating systems and I would like to solve this problem ;-)

**Comment by Jookia** [comment]
If you just need to access the Internet, slirp is the easiest and most reliable way to do this. pcap uses raw network access and is tempermental.

What OS are you running DOSBox-X on?

**Comment by Bedenk** [comment]
I am using dosbox-x on Ubuntu 22.04. And i am using slrip now, it works. No problem there. But about all the howto's on the net explain how to use win98 with dosbox-x and pcap. So it is logic to try that first.

**Comment by rderooy** [comment]
Yes I think we need a separate issue to see what the issue is with PCAP. Try an older version of DOSBox-X to see when it was working and then do a git bisect to find the offending commit.

**Comment by Jookia** [comment]
Basically what @rderooy said. Basically:

- Could you try pcap again
- And open an issue about how it isn't working with your DOSBox-X version

Additionally, try older DOSBox-X versions to see if that helps.
